### PR TITLE
chore: Update Field spec for FieldContext

### PR DESCRIPTION
**Pull request by behowell**
## New Behavior

Update Field's spec.md to describe the new method of passing field state via context instead of setting props directly on the child.

## Related Issue(s)

- #27399

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-checkbox</samp> <br /> <abbr title='bundle-size/Checkbox.fixture.js'>Checkbox</abbr>                                                                              |   `37.02 kB`<br />`11.255 kB` |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Combobox.fixture.js'>Combobox (including child components)</abbr>                                                 |   `89.718 kB`<br />`28.49 kB` |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Dropdown.fixture.js'>Dropdown (including child components)</abbr>                                                 |  `88.402 kB`<br />`28.217 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `67.994 kB`<br />`18.395 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `207.512 kB`<br />`57.555 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `38.819 kB`<br />`12.351 kB` |
| <samp>react-datepicker-compat</samp> <br /> <abbr title='bundle-size/DatePicker.fixture.js'>DatePicker Compat</abbr>                                                          | `251.272 kB`<br />`67.329 kB` |
| <samp>react-field</samp> <br /> <abbr title='bundle-size/Field.fixture.js'>Field</abbr>                                                                                       |   `21.509 kB`<br />`7.408 kB` |
| <samp>react-input</samp> <br /> <abbr title='bundle-size/Input.fixture.js'>Input</abbr>                                                                                       |   `26.707 kB`<br />`8.113 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `6.446 kB`<br />`2.185 kB` |
| <samp>react-progress</samp> <br /> <abbr title='bundle-size/ProgressBar.fixture.js'>ProgressBar</abbr>                                                                        |   `16.322 kB`<br />`5.822 kB` |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/Radio.fixture.js'>Radio</abbr>                                                                                       |   `35.48 kB`<br />`11.126 kB` |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/RadioGroup.fixture.js'>RadioGroup</abbr>                                                                             |   `18.889 kB`<br />`6.929 kB` |
| <samp>react-select</samp> <br /> <abbr title='bundle-size/Select.fixture.js'>Select</abbr>                                                                                    |   `27.915 kB`<br />`9.189 kB` |
| <samp>react-slider</samp> <br /> <abbr title='bundle-size/Slider.fixture.js'>Slider</abbr>                                                                                    |  `36.845 kB`<br />`11.474 kB` |
| <samp>react-spinbutton</samp> <br /> <abbr title='bundle-size/SpinButton.fixture.js'>SpinButton</abbr>                                                                        |  `36.598 kB`<br />`10.752 kB` |
| <samp>react-switch</samp> <br /> <abbr title='bundle-size/Switch.fixture.js'>Switch</abbr>                                                                                    |   `32.438 kB`<br />`9.706 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/DataGrid.fixture.js'>DataGrid</abbr>                                                                                 | `150.829 kB`<br />`41.052 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/PrimitivesOnly.fixture.js'>Table (Primitives only)</abbr>                                                            |  `47.119 kB`<br />`12.703 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableAsDataGrid.fixture.js'>Table as DataGrid</abbr>                                                                 | `138.971 kB`<br />`35.722 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSelectionOnly.fixture.js'>Table (Selection only)</abbr>                                                         |    `86.8 kB`<br />`21.742 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSortOnly.fixture.js'>Table (Sort only)</abbr>                                                                   |   `86.13 kB`<br />`21.549 kB` |
| <samp>react-textarea</samp> <br /> <abbr title='bundle-size/Textarea.fixture.js'>Textarea</abbr>                                                                              |   `30.108 kB`<br />`9.478 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/1071a2a382741b1328228b336c2a94de388a3963'>1071a2a382741b1328228b336c2a94de388a3963</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27454/perf-test-react-components/InfoButton-mount_base.html">24</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27454/perf-test-react-components/InfoButton-mount.html">21</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27454/perf-test-react-components/InfoButton-mount.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27454/perf-test-react-components/Avatar-mount_base.html">671</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27454/perf-test-react-components/Avatar-mount.html">713</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27454/perf-test-react-components/Button-mount_base.html">387</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27454/perf-test-react-components/Button-mount.html">375</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27454/perf-test-react-components/Field-mount_base.html">1251</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27454/perf-test-react-components/Field-mount.html">1264</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27454/perf-test-react-components/FluentProvider-mount_base.html">900</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27454/perf-test-react-components/FluentProvider-mount.html">885</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27454/perf-test-react-components/FluentProviderWithTheme-mount_base.html">110</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27454/perf-test-react-components/FluentProviderWithTheme-mount.html">115</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27454/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">104</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27454/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">89</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27454/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">111</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27454/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">100</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27454/perf-test-react-components/InfoButton-mount_base.html">24</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27454/perf-test-react-components/InfoButton-mount.html">21</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27454/perf-test-react-components/InfoButton-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27454/perf-test-react-components/MakeStyles-mount_base.html">1123</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27454/perf-test-react-components/MakeStyles-mount.html">1131</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27454/perf-test-react-components/Persona-mount_base.html">1979</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27454/perf-test-react-components/Persona-mount.html">1961</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27454/perf-test-react-components/SpinButton-mount_base.html">1591</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27454/perf-test-react-components/SpinButton-mount.html">1604</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

File: packages/react-components/react-field/docs/Spec.md [hunk 1]
```diff
@@ -286,13 +354,15 @@ The Field itself is not interactive. The wrapped component has the same interact
 - **ARIA pattern**
   - Field itself does not implement a defined ARIA pattern. It has no role applied to the root element.
 - **Attributes**
-  - The following are applied on the child component:
+  - The following are applied by `useFieldControlProps` or the child render function:
```
Comment (khmakoto) [inline packages/react-components/react-field/docs/Spec.md:357]: _nit:_ wording

```suggestion
  - The following are applied by `useFieldControlProps` or passed to the child render function:
```

**Review by khmakoto — APPROVED** [review]

**Pull request merged**